### PR TITLE
Qa deploy

### PR DESCRIPTION
**Pull request by danamansana**
This branch modifies the deployment related files. Package.json has new deploy scripts, config has var_[env].env files, redundant variables are removed from .env, and we are no longer using samples. The README has been modified to reflect these changes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -75,23 +75,20 @@ itemOwningInstitution, requestingInstitution
...
   * `brew install php71-pdo-pgsql`
-  
+
 ## Installation
 
 1. Clone the repo.
 2. Install required dependencies.
    * Run `npm install` to install Node.js packages.
    * Run `composer install` to install PHP packages.
    * If you have not already installed `node-lambda` as a global package, run `npm install -g node-lambda`.
```
Comment (nonword) [inline README.md]: can prob remove this line (91)

File: package.json [hunk 2]
```diff
@@ -6,14 +6,15 @@
   "scripts": {
     "pretest": "phplint src/*.php src/**/*.php src/**/**/*.php",
     "test-request": "node-lambda run -j config/event_sources_development.json -f tests/Mocks/mock.env",
-    "deploy-development": "node-lambda deploy -e development -f config/var_development.env -S config/event_sources_development.json -b {subnets} -g {securitygroups}",
-    "deploy-qa": "node-lambda deploy -e qa -f config/var_qa.env -S config/event_sources_qa.json -b {subnets} -g {securitygroups}",
-    "deploy-production": "node-lambda deploy -e production -f config/var_production.env -S config/event_sources_production.json -b {subnets} -g {securitygroups}"
+    "deploy-development": "./node_modules/.bin/node-lambda deploy -e development -f config/var_development.env -S config/event_sources_development.json --role arn:aws:iam::224280085904:role/lambda_basic_execution --profile nypl-sandbox -b subnet-f4fe56af -g sg-1d544067",
```
Comment (nonword) [inline package.json:9]: Does `config/event_sources_development.json` exist?

**Review by nonword — COMMENTED** [review]
Looks good. Couple small comments

File: package.json (hunk 2 above)
Comment (danamansana) [reply package.json:9]: Fixed

File: README.md (hunk 1 above)
Comment (danamansana) [reply README.md]: Yep, fixed

**Comment by danamansana** [comment]
I've fixed the issues raised above and added travis scripts

File: package.json [hunk 3]
```diff
@@ -6,14 +6,15 @@
...
+    "deploy-qa": "./node_modules/.bin/node-lambda deploy -e qa -f config/var_qa.env -S config/event_sources_qa.json --role arn:aws:iam::946183545209:role/lambda-full-access --profile nypl-digital-dev -b subnet-f35de0a9,subnet-21a3b244 -g sg-aa74f1db",
+    "deploy-production": "./node_modules/.bin/node-lambda deploy -e production -f config/var_production.env -S config/event_sources_production.json --role arn:aws:iam::946183545209:role/lambda-full-access --profile  nypl-digital-dev -b subnet-59bcdd03,subnet-5deecd15 -g sg-116eeb60"
   },
   "repository": {},
   "keywords": [],
   "dependencies": {
-    "http-string-parser": "^0.0.5"
+    "http-string-parser": "^0.0.5",
+    "node-lambda": "^0.12.0"
```
Comment (nonword) [inline package.json]: Is `node-lambda` a production dependency?

File: .travis.yml [hunk 4]
```diff
@@ -1,10 +1,37 @@
...
+  - if [ "$TRAVIS_BRANCH" == "master" ]; then ENVIRONMENT_NAME=production; fi
   - vendor/bin/coveralls -v
   - vendor/bin/phpcs -n --standard=PSR1,PSR2 src/
   - vendor/bin/phpcbf src/
+before_deploy:
+- echo "All unit tests passed; Preparing to deploy $TRAVIS_BRANCH"
+deploy:
+- provider: script
+  skip_cleanup: true
+  script: "./scripts/travis-deploy.sh $TRAVIS_BRANCH"
```
Comment (nonword) [inline .travis.yml]: I think you want `"./scripts/travis-deploy.sh $ENVIRONMENT_NAME"`

File: package.json (hunk 3 above)
Comment (danamansana) [reply package.json]: It's necessary for the deployment scripts. Should it not be a production dependency?

File: .travis.yml (hunk 4 above)
Comment (danamansana) [reply .travis.yml]: Fixed

File: package.json (hunk 3 above)
Comment (nonword) [reply package.json]: I believe travis installs dependencies via `npm install` to run tests, so should have all necessary dependencies (dev and production). Anything not actually required by application code should be in `devDependencies`. Installing *all* dependencies is the default; You have to install via `npm install --production` to restrict to production dependencies (which happens later via `node-lambda package` to build the deployment for upload).

Comment (danamansana) [reply package.json]: Fixed

**Pull request merged**